src/Compilers/CSharp/Test/Syntax/Parsing/LocalFunctionParsingTests.cs

```diff
@@ -415,21 +415,6 @@ public void NeverEndingTest()
                 // (3,22): error CS1513: } expected
                 //         async public virtual M() {}
                 Diagnostic(ErrorCode.ERR_RbraceExpected, "virtual").WithLocation(3, 22),
-                // (3,30): error CS1520: Method must have a return type
```

Comment:
Have you replicated this test over in binding to make sure that the same errors are caught?